gerrit_changes_to_github.py

```diff
@@ -271,6 +183,82 @@ def is_repository_usable(args):
     return got_gerrit and got_target
 
 
+def get_changes_from_event_logs(args, gerrit_password, ghpa_token):
+    """Go through all events since the latest timestamp"""
+    try:
+        latest_timestamp = int(os.getenv("LAST_TIMESTAMP"))
+    except Exception as e:
+        print(f"Error reading last timestamp: {e}")
+
+    latest_ts_as_date = datetime.fromtimestamp(
+        latest_timestamp, tz=timezone.utc
+    ).strftime("%Y-%m-%d %H:%M:%S")
+    log.info(f"Using last timestamp (UTC): {latest_ts_as_date}")
+
+    if (
+        events := requests.get(
+            f"https://review.spdk.io/a/plugins/"
+            f"events-log/events/?t1={latest_ts_as_date}",
+            auth=("spdk-community-ci-samsung", gerrit_password),
+        )
+    ).status_code != 200:
+
+        log.error(f"Failed to retrieve events. HTTP Status Code: {events.status_code}")
+        return None
+
+    events = events.text.replace("}{", "},{").replace("}\n{", "},{")
+    events = json.loads(f"[{events}]")
+
+    refs = []
+    for event in events:
+        event_ts = event.get("eventCreatedOn")
+        latest_timestamp = event_ts if event_ts > latest_timestamp else latest_timestamp
```

Comment:
are the events always returned in chronological order?

if so, then you should be able to just get the latest timestamp from the last event, after the loop is done, rather than updating it every time through the loop

Reply:
Resolved during our meeting: To avoid getting a fraction of a second of events (ex: multiple events have the same creation time and we make a query in the middle of a second, getting half the events for that second), we change the query to have t1=start_time and t2=now-1 to get all the events from t1 to t2.

Then we can just update latest_timestamp with t2.